src/validate/validate_state.test.ts

```diff
@@ -0,0 +1,149 @@
+import {validateState} from './validate_state';
+
+describe('Validate state', () => {
+    test('Should return error if type is not an object', () => {
```

Comment:
Please split the test into several test and use the test name to describe the different scenarios.

Reply:
Those tests were redundant, I removed them.